src/chart/bar/BarChart.tsx

```diff
@@ -203,80 +236,174 @@ const NeoBarChart = (props: ChartProps) => {
+
+  // Container to make the chart scroll horizontally
+  const scrollableWrapperStyle: React.CSSProperties = {
+    width:
+      legendPosition === 'Horizontal'
+        ? adaptableWidth > itemWidthConst * data.length + 200
+          ? adaptableWidth
+          : itemWidthConst * data.length + 200
+        : barWidth * 5 * data.length + itemWidthConst,
+    height: expandForLegend ? 18 * data.length + itemWidthConst * 1.2 + marginBottom : '100%',
+    whiteSpace: 'nowrap',
+  };
+
+  // Container for scrolling container to scroll in
+  const barChartStyle: React.CSSProperties = {
+    width: '100%',
+    overflowX: 'auto',
+    overflowY: 'auto',
+    height: '100%',
+  };
+
   const chart = (
-    <BarChartComponent
-      theme={canvas ? themeNivoCanvas(props.theme) : themeNivo}
-      data={data}
-      key={`${selection.index}___${selection.value}`}
-      layout={layout}
-      groupMode={groupMode == 'stacked' ? 'stacked' : 'grouped'}
-      enableLabel={enableLabel}
-      keys={keys}
-      indexBy='index'
-      margin={{
-        top: marginTop,
-        right: legend ? legendWidth + marginRight : marginRight,
-        bottom: marginBottom,
-        left: marginLeft,
-      }}
-      valueScale={{ type: valueScale }}
-      padding={0.3}
-      minValue={minValue}
-      maxValue={maxValue}
-      colors={getBarColor}
-      axisTop={null}
-      axisRight={null}
-      axisBottom={{
-        tickSize: 5,
-        tickPadding: 5,
-        tickRotation: labelRotation,
-      }}
-      axisLeft={{
-        tickSize: 5,
-        tickPadding: 5,
-        tickRotation: 0,
-      }}
-      labelSkipWidth={labelSkipWidth}
-      labelSkipHeight={labelSkipHeight}
-      labelTextColor={{ from: 'color', modifiers: [['darker', 1.6]] }}
-      {...extraProperties}
-      legends={
-        legend
-          ? [
-              {
-                dataFrom: 'keys',
-                anchor: 'bottom-right',
-                direction: 'column',
-                justify: true,
-                translateX: 120,
-                translateY: 0,
-                itemsSpacing: 2,
-                itemWidth: legendWidth - 28,
-                itemHeight: 20,
-                itemDirection: 'right-to-left',
-                itemOpacity: 0.85,
-                symbolSize: 20,
-                effects: [
-                  {
-                    on: 'hover',
-                    style: {
-                      itemOpacity: 1,
+    <div style={barChartStyle}>
+      <div style={scrollableWrapperStyle}>
+        <BarChartComponent
+          theme={canvas ? themeNivoCanvas(props.theme) : themeNivo}
+          data={data}
+          key={`${selection.index}___${selection.value}`}
+          layout={layout}
+          groupMode={groupMode == 'stacked' ? 'stacked' : 'grouped'}
+          enableLabel={enableLabel}
+          onClick={(e) => {
+            // Get the original record that was used to draw this bar (or a group in a bar).
```

Comment:
is this `onClick` method the same as the `handleBarClick` method above?
If yes can we join them?
Or at least extract this `onClick` out of the react component into its own seperate function?

Reply:
Done